forgit.plugin.zsh

```diff
@@ -190,8 +190,16 @@ forgit::rebase() {
         --preview=\"$preview\"
         $FORGIT_REBASE_FZF_OPTS
     "
-    commit=$(eval "$cmd" | FZF_DEFAULT_OPTS="$opts" fzf | eval "$forgit_extract_sha")
-    [[ -n "$commit" ]] && git rebase -i "$commit"
+    target_commit=$(eval "$cmd" | FZF_DEFAULT_OPTS="$opts" fzf | eval "$forgit_extract_sha")
+    if [[ -n "$target_commit" ]]; then
+        # "$target_commit~" is invalid when the commit is the first commit, but we can use "--root" instead
+        if [[ "$(git rev-parse "$target_commit")" == "$(git rev-list --max-parents=0 HEAD)" ]]; then
+            prev_commit="--root"
+        else
+            prev_commit="$target_commit~"
+        fi
```

Comment:
How about abstracting this into a function named `forgit_previous_commit` so that we can reuse it in `rebase` and `fixup`?

Reply:
Totally makes sense. Give me another few minutes.

Reply:
And there it is!